content-security-policy/script-src/script-src-strict_dynamic_and_unsafe_eval_eval.html

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>Scripts injected via `eval` are allowed with `strict-dynamic` with `unsafe-eval`.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+    <script src='../support/checkSecurityPolicyViolations.sub.js' nonce='dummy'></script>
+    <!-- CSP served: script-src 'strict-dynamic' 'nonce-dummy' 'unsafe-eval' -->
+</head>
+
+<body>
+    <h1>Scripts injected via `eval` are allowed with `strict-dynamic` with `unsafe-eval`.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+        var evalScriptRan = false;
+        async_test(function(t) {
+            try {
+                eval("evalScriptRan = true;");
+            } catch (e) {
+                assert_unreached("`eval` should be allowed with `strict-dynamic` with `unsafe-eval`.");
+            }
+            assert_true(evalScriptRan);
+            t.done();
+        }, "Script injected via `eval` is allowed with `strict-dynamic` with `unsafe-eval`.");
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            setTimeout(t.step_func_done(function() {
+                assert_no_violations();
+            }), 1000);
```

Comment:
A second is a long time to wait for automated tests. Since there's no way to prove that the event _never ever_ fires, I'd suggest that it's enough to add something like `document.addEventListener('securitypolicyviolation', t.unreached_func("No violation event should fire."));` to the first test.